ksql-parser/src/main/java/io/confluent/ksql/parser/tree/HoppingWindowExpression.java

```diff
@@ -116,4 +119,9 @@ public KTable applyAggregate(
             .advanceBy(advanceByUnit.toMillis(advanceBy))
     ).aggregate(initializer, aggregator, materialized);
   }
+
+  @Override
+  public <K> Serde<Windowed<K>> getKeySerde(final Class<K> innerType) {
+    return WindowedSerdes.timeWindowedSerdeFrom(innerType);
+  }
```

Comment:
We should not have any dependency to the physical layer in the parser. This need to be moved to the engine package. See #1965

Reply:
OK, I'll take a look.

Reply:
I've had a look now - I agree this doesn't belong here. But I propose we leave it hear for now and address how/where to move it in #1965.  Trying to move it out now just means hacking in an if/else statement somewhere. Whereas we should fix it using the approach that #1965 finally settles on.

Reply:
Sounds good!